impl/src/main/java/org/apache/myfaces/view/facelets/PartialStateManagementStrategy.java

```diff
@@ -642,14 +642,25 @@ public void invokeContextCallback(FacesContext context, UIComponent target)
                 {
                     try
                     {
-                        target.getChildren().add(childIndex, child);
+                        UIComponent parent = target;
+                        UIComponent dup = parent.getChildren().get(childIndex);
+                        if (child.getId().equals(dup.getId())) // avoid DuplicateIdException  - myfaces-4623
```

Comment:
Maybe I should check for `dup.getAttributes().get(ComponentSupport.MARK_CREATED)`?

Reply:
Actually, I cannot because of the initial check: `child.getAttributes().containsKey(ComponentSupport.MARK_CREATED)` is false.